src/js/angular/import/controllers.js

```diff
@@ -143,30 +144,37 @@ importViewModule.controller('ImportViewCtrl', ['$scope', 'toastr', '$interval',
             };
 
             if (GuidesService.isActive()) {
-                // Prevents closing dialog when user click outside the if.
+                // Prevents closing dialog when user click outside the dialog.
                 options.backdrop = 'static';
                 options.keyboard = false;
             }
 
-            const modalInstance = $uibModal.open(options);
-            modalInstance.result.then(function (settings) {
-                $scope.settings = settings;
-                if ($scope.settingsFor === '') {
-                    $scope.importSelected();
-                } else {
-                    $scope.importFile($scope.settingsFor, true);
-                }
-
-            }, function (settings) {
-                $scope.settings = settings;
-            });
+            $uibModal.open(options).result.then(
+                (settings) => {
+                    $scope.settings = settings;
+                    if ($scope.settingsFor === '') {
+                        $scope.importSelected();
+                    } else {
+                        $scope.importFile($scope.settingsFor, true);
+                    }
+                },
+                (settings) => {
+                    $scope.settings = settings;
+                    onImportRejectHandler && onImportRejectHandler();
```

Comment:
Doesn't sonar complain about this kind of assesments? In other projects it does.

Reply:
It does. It suggests optional chaining instead. I'll change it.

Reply:
Using an optional chaining would also show as a code smell.
You should check with an `if` statement

Reply:
> Using an optional chaining would also show as a code smell.

Really?! Why they are suggesting it then? :)
Anyway. I did it with an if clause because I can't do the optional chaining in this case.